### PR TITLE
Introduce unit-tests for issues fixed in v8

### PR DESCRIPTION
**Pull request by martintmk**
### Details on the issue fix or feature implementation

Here we add unit tests for issues addressed in v8. This demonstrates the issue reporters the problem is fixed in v8.

### Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

**Comment by martintmk** [comment]
I think we can just merge this PR and keep the branch. It just adds more tests which is useful to have in main.

File: src/Polly.Extensions.Tests/Issues/IssuesTests.OnCircuitBreakWithServiceProvider_796.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
+using Microsoft.Extensions.DependencyInjection;
+using Polly.CircuitBreaker;
+using Polly.Extensions.DependencyInjection;
+using Polly.Registry;
+
+namespace Polly.Core.Tests.Issues;
+
+public class IssuesTests
```
Comment (martincostello) [inline src/Polly.Extensions.Tests/Issues/IssuesTests.OnCircuitBreakWithServiceProvider_796.cs:8]: Shall we make this partial too so we don't need to add it later?

Comment (martintmk) [reply src/Polly.Extensions.Tests/Issues/IssuesTests.OnCircuitBreakWithServiceProvider_796.cs:8]: I tried but it complains about https://rules.sonarsource.com/csharp/RSPEC-2333.

Should be no problem to just add it later when we add more issues.

**Review by martincostello — APPROVED** [review]

File: src/Polly.Extensions.Tests/Issues/IssuesTests.OnCircuitBreakWithServiceProvider_796.cs (hunk 1 above)
Comment (martintmk) [reply src/Polly.Extensions.Tests/Issues/IssuesTests.OnCircuitBreakWithServiceProvider_796.cs]: @martincostello I just tried to add empty `IssuesTests` class as in other cases. It worked so it's non issue anyway.

**Review by martincostello — APPROVED** [review]

**Pull request merged**